core/src/main/java/org/apache/brooklyn/core/catalog/internal/JavaCatalogToSpecTransformer.java

```diff
@@ -68,11 +73,11 @@ public boolean accepts(String planType) {
     public <T, SpecT extends AbstractBrooklynObjectSpec<? extends T, SpecT>> SpecT createCatalogSpec(
             CatalogItem<T, SpecT> item, Set<String> encounteredTypes) throws PlanNotRecognizedException {
         if (item.getJavaType() != null) {
-            log.warn("Deprecated functionality (since 0.9.0). Using old-style xml catalog items with java type attribute for " + item);
```

Comment:
`getJavaType() != null` should still be deprecated for catalog items, even if introduced in the type registry. It is non-null only for old-style xml catalog files.